File: src/utils/logger.js
const util = require('util');

const winston = require('winston');
require('winston-daily-rotate-file');

const maxFileSize = process.env.MAX_LOG_FILE_SIZE || "100m";
const maxLogFiles = process.env.MAX_LOG_FILES || "50";

var transport = new winston.transports.DailyRotateFile({
  level: "info",
  filename: "./logs/r1x-%DATE%.log",
  datePattern: "YYYY-MM-DD",
  zippedArchive: true,
  maxSize: maxFileSize,
  maxFiles: maxLogFiles
});

var logger = winston.createLogger({
  format: winston.format.combine(
    winston.format.timestamp({format: 'YYYY-MM-DD HH:mm:ss.SSS'}),
    winston.format.printf((info) => {
      return `${info.timestamp} ${info.message}`;
    })
  ),
  transports: [
    transport,
    new winston.transports.Console()
  ]
});

function createLoggingContext(context) {
  const logFn = function log(message, ...args) {
    const inspectedArgs = [util.inspect(message)]
    for (const arg of args) {
      inspectedArgs.push(util.inspect(arg));
    }
 
    mergedMessage = `[${context}] ${inspectedArgs.join(' ')}`;
    logger.info(mergedMessage);
  }

  return { log : logFn };
} 

module.exports = {
  logger,
  createLoggingContext
};

File: src/utils/logger.py
import os
import logging
from logging.handlers import TimedRotatingFileHandler

max_file_size = os.environ.get("MAX_LOG_FILE_SIZE", 100 * 1024 * 1024)
max_log_files = int(os.environ.get("MAX_LOG_FILES", 50))

log_formatter = logging.Formatter('%(asctime)s %(message)s', datefmt='%Y-%m-%d %H:%M:%S.%f')

file_handler = TimedRotatingFileHandler('./logs/r1x.log', when='D', interval=1, backupCount=max_log_files)
file_handler.setFormatter(log_formatter)
file_handler.setLevel(logging.INFO)
file_handler.suffix = '%Y-%m-%d'
file_handler.extMatch = file_handler.extMatch

console_handler = logging.StreamHandler()
console_handler.setFormatter(log_formatter)
console_handler.setLevel(logging.INFO)

logger = logging.getLogger()
logger.setLevel(logging.INFO)
logger.addHandler(file_handler)
logger.addHandler(console_handler)

def create_logging_context(context):
    def log_fn(message, *args):
        merged_message = f"[{context}] {message} {' '.join(str(arg) for arg in args)}"
        logger.info(merged_message)

    return {"log": log_fn}

